source/kernel/src/drivers/emb_display.rs

```diff
@@ -0,0 +1,260 @@
+        let commander = CommanderTask {
+            cmd: cmd_cons,
+            fmutex: imutex,
+        };
+
+        kernel.spawn(async move{
+            commander.run().await;
+        }).await;
+
+        kernel
+            .with_registry(|reg| reg.register_konly::<EmbDisplay>(&cmd_prod))
+            .await
+            .map_err(|_| RegistrationError::DisplayAlreadyExists).unwrap();
+
+        Ok(())
+    }
+}
+
+// Create and return a Simulator display object from raw pixel data.
+// Pixel data is turned into a raw image, and then drawn onto a SimulatorDisplay object
+// This is necessary as a e-g Window only accepts SimulatorDisplay object
+// On a physical display, the raw pixel data can be sent over to the display directly 
+// Using the display's device interface
+impl FrameChunk {
+    pub async fn frame_display(&mut self) -> Result<ImageRaw<Gray8>, ()> {
+        let raw_image: ImageRaw<Gray8>;
+        raw_image = ImageRaw::<Gray8>::new(self.bytes.as_ref(), self.width);
+        Ok(raw_image)
+    }
+
+    pub async fn frame_clear(&mut self) {
+        for elem in self.bytes.iter_mut() { *elem = 0; }
+        
+    }
+}
+
+impl EmbDisplayHandle {
+    pub async fn from_registry(kernel: &'static Kernel) -> Option<Self> {
+        let prod = kernel.with_registry(|reg| reg.get::<EmbDisplay>()).await?;
+
+        Some(EmbDisplayHandle {
+            prod,
+            reply: Reusable::new_async(kernel).await,
+        })
+    }
+
+    pub async fn get_framechunk(&mut self, frame_id: u16, start_x: i32, start_y: i32, width: u32, height: u32) -> Option<FrameChunk> {
+        self.prod
+            .send(
+                Request::NewFrameChunk { frame_id, start_x, start_y, width, height },
+                ReplyTo::OneShot(self.reply.sender().ok()?),
+            )
+            .await
+            .ok()?;
+
+        let resp = self.reply.receive().await.ok()?;
+        let body = resp.body.ok()?;
+
+        let Response::FrameChunkAllocated(frame) = body;
+        Some(frame)
+    }
+}
+
+impl DisplayInfo {
+    // Returns a new frame chunk
+    async fn new_frame(
+        &mut self,
+        frame_id: u16,
+        start_x: i32,
+        start_y: i32,
+        width: u32,
+        height: u32,
+    ) -> Result<FrameChunk, FrameError> {
+        if self.frames.is_full() {
+            return Err(FrameError::RegistryFull);
+        }
+
+        if self.frames.iter().any(|f| f.frame == frame_id) {
+            return Err(FrameError::DuplicateItem);
+        }
+
+        self.frames.push(FrameInfo { frame: frame_id })
+            .map_err(|_| FrameError::RegistryFull)?;
+
+        let size = (width * height) as usize;
+        let bytes = self.kernel.heap().allocate_array_with(|| 0, size).await;
+        let fc = FrameChunk {
+            frame_id: frame_id,
+            bytes: bytes,
+            start_x: start_x,
+            start_y: start_y,
+            width: width,
+            height: height,
```

Comment:
style nit, not a big deal: we can use [field initializer shorthand](https://rust-lang.github.io/rfcs/1682-field-init-shorthand.html) here, which maybe looks a little nicer:
```suggestion
            frame_id
            bytes,
            start_x,
            start_y,
            width,
            height,
```

Reply:
This does look  a lot better!